src/main/java/com/BlogSite/TestBlogProject/services/UserServiceImpl.java

```diff
@@ -24,38 +24,36 @@ public List<User> getUsers() {
     public Result<User> getUser(Long id) {
         Result<User> result = new Result<>();
         User user = userRepository.findById(id).orElse(null);
-        if(user == null) {
+        if (user == null)
             result.setError(ErrorCode.USER_NOT_FOUND);
-            return result;
-        }
-        result.setData(user);
+        else
+            result.setData(user);
         return result;
     }
 
     @Override
     public Result<User> getUserByUsername(String username) {
         Result<User> result = new Result<>();
         User user = userRepository.findByUsername(username).orElse(null);
-        if(user == null) {
+        if (user == null)
             result.setError(ErrorCode.USER_NOT_FOUND);
-            return result;
-        }
-        result.setData(user);
+        else
+            result.setData(user);
         return result;
     }
 
     @Override
     public Result<User> addUser(UserDto userDto) {
         Result<User> result = new Result<>();
-        if (userRepository.findByUsername(userDto.getUsername()).isPresent()) {
+        if (userRepository.findByUsername(userDto.getUsername()).isPresent())
```

Comment:
if you wanna use lambda style you could replace this if with Optional.ifPresent(t->...)